src/Microsoft.ML.AutoML/AutoMLExperiment/IPerformanceMonitor.cs

```diff
@@ -122,5 +132,17 @@ private void SampleCpuAndMemoryUsage()
                 PerformanceMetricsUpdated?.Invoke(this, metrics);
             }
         }
+
+        public virtual void OnPerformanceMetricsUpdatedHandler(TrialSettings trialSettings, TrialPerformanceMetrics metrics, CancellationTokenSource trialCancellationTokenSource)
+        {
+            if (_settings.MaximumMemoryUsageInMegaByte is double d && metrics.PeakMemoryUsage > d && !trialCancellationTokenSource.IsCancellationRequested)
```

Comment:
You can bring [ReportTrialResourceUsage](https://github.com/andrasfuchs/BioBalanceDetector/blob/6636da3f7c512aff9dc1c562cbea2d6320a742c5/Software/Source/BBDProto08/BBD.BodyMonitor/BBD.BodyMonitor.API/Models/MLContextLoggerMonitor.cs#L59) here. It will be called every time a new PerformanceMetric get sampled